perllib/FixMyStreet/App/Controller/Report/New.pm

```diff
@@ -744,6 +744,24 @@ sub setup_categories_and_bodies : Private {
     my $contacts = $c->model('DB::Contact')->for_new_reports($c, \%bodies);
     my @contacts = $c->cobrand->categories_restriction($contacts)->all_sorted;
 
+    # If there are multiple contacts with the same category name and one of
+    # them has prefer_if_multiple set then use that one.
+    my %contacts_by_category;
+    foreach (@contacts) {
+        push @{$contacts_by_category{$_->category}}, $_;
+    }
+    @contacts = grep {
+        my $category = $_->category;
+        if (scalar @{$contacts_by_category{$category}} > 1 && any { $_->get_extra_metadata('prefer_if_multiple') } @{$contacts_by_category{$category}}) {
```

Comment:
I don't think you need the size check, which helps (if it's size one and there is a preferred one, it'll match), and other thing I can think of is to pre-fetch the 'preferred' contact IDs and categories. Does this then work the same?

```
my @preferred = grep { $_->get_extra_metadata('prefer_if_multiple') } @contacts;
my %preferred_ids = map { $_->id => 1 } @preferred;
my %preferred_cats = map { $_->category => 1 } @preferred;

@contacts = grep { !$preferred_cats{$_->category} || $preferred_ids{$_->id} } @contacts;
```

Reply:
Ah that's much nicer, and seems to work the same yep!

Have pushed a fixup to my fixup.